HLTrigger/special/src/HLTRechitsToDigis.cc

```diff
@@ -0,0 +1,204 @@
+  using namespace edm;
+  // handles for digis
+  Handle<EBDigiCollection> digisEBHandle;
+  Handle<EEDigiCollection> digisEEHandle;
+  
+  // output collections
+  std::auto_ptr<EBDigiCollection> outputEBDigiCollection( new EBDigiCollection );
+  std::auto_ptr<EEDigiCollection> outputEEDigiCollection( new EEDigiCollection );
+  
+  // calibrated rechits
+  Handle<EcalRecHitCollection> recHitsHandle;
+  iEvent.getByToken(recHitsToken_, recHitsHandle);   
+  
+  // match the digis based on the region
+  switch(regionID_) {
+  case 1: { // ECAL EB = 1
+    iEvent.getByToken(digisEBInToken_, digisEBHandle);
+    const EBDigiCollection* digisEB = digisEBHandle.product();   
+    
+    // loop over the collection of rechits and match to digis
+    EcalRecHitCollection::const_iterator ituneEB;
+    for (ituneEB = recHitsHandle->begin(); ituneEB != recHitsHandle->end(); ituneEB++) {
+      EcalRecHit hit = (*ituneEB);
+      outputEBDigiCollection->push_back( (*digisEB->find(hit.id())).id(), (*digisEB->find(hit.id())).begin() );
+    }
+    
+    iEvent.put( outputEBDigiCollection, digisOut_);     
+    break;   
+  }
+    
+  case 2: {  // ECAL EE = 2
+    iEvent.getByToken(digisEEInToken_, digisEEHandle);
+    const EEDigiCollection* digisEE = digisEEHandle.product();   
+    
+    // loop over the collection of rechits and match to digis
+    EcalRecHitCollection::const_iterator ituneEE;
+    for (ituneEE = recHitsHandle->begin(); ituneEE != recHitsHandle->end(); ituneEE++) {
+      EcalRecHit hit = (*ituneEE);      
+      outputEEDigiCollection->push_back( (*digisEE->find(hit.id())).id(), (*digisEE->find(hit.id())).begin() );              
+    }
+    
+    iEvent.put(outputEEDigiCollection, digisOut_);     
+    break;
+  }
+  } 
+}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+HLTRechitsToDigis::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+HLTRechitsToDigis::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+HLTRechitsToDigis::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+HLTRechitsToDigis::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+HLTRechitsToDigis::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+
+  desc.add<uint32_t>("regionID_1EB_2EE", 1);
```

Comment:
please put a description of the possible parameters in the comments for the parameter.